client.go

```diff
@@ -67,6 +68,20 @@ func Dial(pconn net.PacketConn, remoteAddr net.Addr, host string, config *Config
 	return c.establishConnection()
 }
 
+func populateClientConfig(config *Config) *Config {
+	versions := config.Versions
+	if len(versions) == 0 {
+		versions = make([]protocol.VersionNumber, len(protocol.SupportedVersions))
```

Comment:
Because I don't want to allow runtime-changes of the supported versions. Doing so will likely lead to nasty race conditions (clients won't like it if a VNP contains a certain set of versions, but the SHLO contains a different one).
The cost for copying this slice is minimal, but makes it explicit what is going on (and that runtime changes are not possible).